### PR TITLE
feat(rpc): Add method removal functionality for RPC transports

### LINKED ISSUE #9798: Support replacing rpc handlers
### Describe the feature

ref https://github.com/paradigmxyz/reth/discussions/8294

this will be possible once we bump jsonrpsee and can be integrated into the RpcContext type

https://github.com/paradigmxyz/reth/blob/b864185f6e3115c51e4f44aced04b78b801f8ada/crates/node/builder/src/rpc.rs#L137-L139

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by SkandaBhat**
Fixes #9798 

This PR introduces new methods to remove RPC methods from specific 
transports (HTTP, WebSocket, IPC) or from all transports simultaneously:

- remove_http_method: Removes a method from the HTTP transport
- remove_ws_method: Removes a method from the WebSocket transport
- remove_ipc_method: Removes a method from the IPC transport
- remove_method_from_all_transports: Removes a method from all configured transports

Unit tests have been added to verify the correct functionality of these new methods.

### CODE REVIEW

File: crates/rpc/rpc-builder/src/lib.rs [hunk 1]
```diff
@@ -1705,6 +1705,50 @@ impl TransportRpcModules {
...
+            ipc_module.remove_method(method_name).is_some()
+        } else {
+            false
+        }
+    }
+
+    /// Removes the method with the given name from all configured transports.
+    ///
+    /// Returns `true` if the method was found and removed, `false` otherwise.
+    pub fn remove_method_from_all_transports(&mut self, method_name: &'static str) -> bool {
```
Comment (mattsse) [inline crates/rpc/rpc-builder/src/lib.rs:1745]:
```suggestion
    pub fn remove_method_from_configured(&mut self, method_name: &'static str) -> bool {
```

File: crates/rpc/rpc-builder/src/lib.rs [hunk 2]
```diff
@@ -1705,6 +1705,50 @@ impl TransportRpcModules {
         self.merge_ipc(other)?;
         Ok(())
     }
+
+    /// Removes the method with the given name from the configured http methods.
+    ///
+    /// Returns `true` if the method was found and removed, `false` otherwise.
+    pub fn remove_http_method(&mut self, method_name: &'static str) -> bool {
```
Comment (mattsse) [inline crates/rpc/rpc-builder/src/lib.rs:1712]: these need the disclaimer from `remove_method`, we can reference this function [`RpcModule::remove_method`]

> Be aware that a subscription consist of two methods, subscribe and unsubscribe and it's the caller responsibility to remove both subscribe and unsubscribe methods for subscriptions.

**Review by mattsse — CHANGES_REQUESTED** [review]
cool, naming and doc nit

File: crates/rpc/rpc-builder/src/lib.rs (hunk 2 above)
Comment (SkandaBhat) [reply crates/rpc/rpc-builder/src/lib.rs:1712]: Makes sense, will add!

Comment (SkandaBhat) [reply crates/rpc/rpc-builder/src/lib.rs:1716]: Resolved in 93a510aa620cb50134a9275842b95a3ffbe6aa9f

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
needs fmt

**Comment by mattsse** [comment]
btw if you're looking for something else ptal #10494 :)

**Pull request merged**